app/services/cloud_front_header_parser.rb

```diff
@@ -10,6 +10,7 @@ def client_port
 
   # Source IP and port for client connection to CloudFront
   def viewer_address
+    return nil unless @request&.headers # guard against request being nil
     @request.headers['CloudFront-Viewer-Address']
```

Comment:
maybe we can one-line this
```suggestion
    @request&.headers && @request.headers['CloudFront-Viewer-Address']
```

Reply:
I'm totally open to discussion but am not sure I'm seeing the advantage here.

Over time I've been gravitating towards writing dead-simple Ruby when possible, perhaps to a fault. My thinking with how this is written here is that someone just learning Ruby could read these two lines and confidently understand what's happening and what's returned: nil if there aren't request headers, otherwise the value of a hash key.

While it's not like it's a super-complex logic puzzle, the one-liner requires a little bit more knowledge to understand why it's not returning a boolean, and then why it returns nil, not false, if request is nil. I certainly wouldn't object to this if it were written, but I don't see an obvious advantage to squashing this into one line.

Reply:
fair enough! we can leave as-is